modules/ROOT/pages/testing-oauth-modules.adoc

```diff
@@ -0,0 +1,77 @@
+= OAuth-Enabled Modules
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Mule runtime engine supports authorization through various OAuth grant types, including the Authorization Code grant type, which allows an external system (for example, your Mule application) to operate on behalf of a user without requiring user authentication.
+
+However, because the Authorization Code grant type is designed to force human intervention in the authentication process that identifies users using OAuth (OAuth dance), building an automated test for an application using this grant type is challenging.
+
+To overcome this challenge, use the Storage Event processor to bypass the OAuth dance by manually providing a valid access token. Ensure the following requirements are met:
+
+* You must manually obtain the token.
+* You must be aware that the token eventually expires. +
+How often the token expires depends on the service provider to which you are connecting.
```

Comment:
You can end sentences with a preposition now! "service provider you are connecting to"